vm/src/dictdatatype.rs

```diff
@@ -557,10 +557,13 @@ impl<T: Clone> Dict<T> {
         let removed = if entry_index == inner.used {
             inner.entries.pop().unwrap()
         } else {
-            let last_index = inner.entries.last().unwrap().index;
-            let removed = inner.entries.swap_remove(entry_index);
-            inner.indices[last_index] = entry_index as i64;
-            removed
+            let entries_rem = entry_index + 1;
+            let entries_len = inner.entries.len();
+            for move_entry_idx in entries_rem..entries_len {
+                let index_index = inner.entries[move_entry_idx].index;
+                inner.indices[index_index] = move_entry_idx as i64 - 1;
+            }
```

Comment:
I don't think CPython uses loop for this. What happens if the dict has very many items? Would you check `_PyDict_Pop_KnownHash` in `dictobject.c`? I think this is the corresponding funciton in CPython.

Reply:
Yeah, this definitely won't be a loop, pretty sure that leads to `O(N)` deletes.

Reply:
@eldpswp99 is working on this again